### PR TITLE
docs: add manual tls instructions

### PR DESCRIPTION
**Pull request by loshz**
### Issue

https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2015

### Description

As mentioned in the above issue, it is possible to use self-signed certificates instead of using cert-manager. This PR adds documentation for configuring the AWS LB Controller in this fashion.

I have tested this on k8s 1.20 and 1.21 with controller versions v2.2.0 and v2.2.1.

**Notes:**
- ~I tried to match the doc styling as best as possible. Are there instructions for running the docs locally so I can test?~
- ~I didn't add exact instructions for creating the certs as I feel like it should be up to the user - thoughts?~
- Would be great if someone else could also test the controller with self-signed certs.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by syscll** [comment]
Added some basic `openssl` commands to the best of my knowledge.

File: docs/deploy/installation.md [hunk 1]
```diff
@@ -110,17 +110,56 @@ curl -o iam-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-lo
...
+    ```
+    aws-load-balancer-webhook-service.kube-system.svc
+    aws-load-balancer-webhook-service.kube-system.svc.cluster.local
+    ```
+    
+	2. Create a self-signed certificate from the above certificate authority for the above request.
+
+	3. Once the certs have been created, you need to make them available to the cluster. This can be by create the required `aws-load-balancer-webhook-tls` Kubernetes secret in the `kube-system` namespace:
+    ```
+    kubectl create secret --namespace=kube-system tls aws-load-balancer-webhook-tls \
```
[the remark this answers is not in the document]
Comment (syscll) [reply docs/deploy/installation.md:153]: Have checked versions 1.18-1.22 and this flag doesn't exist for `kubectl create secret tls`

File: docs/deploy/installation.md [hunk 2]
```diff
@@ -110,17 +110,56 @@ curl -o iam-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-lo
...
+    ```
+    apiVersion: admissionregistration.k8s.io/v1
+    kind: MutatingWebhookConfiguration
+    metadata:
+      name: aws-load-balancer-webhook
+    webhooks:
+    - admissionReviewVersions:
+      - v1beta1
+      clientConfig:
+        # PEM encoded CA bundle, created earlier, which will be used to validate the webhook's server certificate.
```
[the remark this answers is not in the document]
Comment (syscll) [reply docs/deploy/installation.md:168]: After looking through the [docs](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/), I can see 4 reference examples of using the `clientConfig.caBundle`.

For `admissionregistration.k8s.io/v1`, 3 of them specify PEM format and [1](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#service-reference) specifies base64-encoded PEM format.

I have tried a plain text PEM formatted string on k8s 1.20-1.22 and it works fine, so I'm unsure which approach we should recommend here?

**Comment by syscll** [comment]
Does anyone have have time to review this? Much appreciated _(not passive-aggressive)_.

File: docs/deploy/installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kishorj) [reply docs/deploy/installation.md:223]: with YAML or JSON spec, the PEM encoded CA bundle must be the base64 encoded PEM certificate including the PEM header and footer. I'm unsure how you verified with the PEM format, whether you stripped the header and footer and input the remaining string. If so, you would be able to edit the webhook configuration, but would get `x509: certificate signed by unknown authority` error later.

We can clarify it in the comments, for example
```
        # PEM encoded CA bundle, created earlier, which will be used to validate the webhook's server certificate. For example: cat aws_lbc.crt | base64
```

File: docs/deploy/installation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kishorj) [reply docs/deploy/installation.md:208]: I'm resolving this conversation.

**Comment by kishorj** [comment]
@syscll, 
I feel the `clientConfig` section in step 4 is confusing. We need to clarify it further so users don't need to spend time trying to debug failures.

File: docs/deploy/installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (syscll) [reply docs/deploy/installation.md:223]: I agree. Will update

[the remark this answers is not in the document]
Comment (syscll) [reply docs/deploy/installation.md:223]: Added your comment. Let me know if I could better document this.

**Comment by rdubya16** [comment]
Would be very nice is someone merged this, took me a while to figure this out. We would prefer not to add cert-manager to our cluster due to managing yet another component, just to get ingress working which didnt even require this for 1.x versions.

**Comment by loshz** [comment]
I just fixed a [bug](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/2298/commits/9f5dd795d75b9b9ce917b60d6fac99142aafbc4b) where I wasn't using the Root CA during self-signed cert creation.

**Comment by loshz** [comment]
Any thoughts on merging this?

**Pull request closed without merging**